### PR TITLE
Dbvt memmove fix

### PR DESCRIPTION
**Pull request by antarespilot**
Hi Erwin,

I finally found time to look into the memmove issue described at https://github.com/bulletphysics/bullet3/issues/347

Here is my proposed fix. I've measured the performance using the google benchmark utility, and the memmove version is always significantly faster than the for-loop (magnitude of app. 4x or more). Of course, this is just an isolated test of the for-loop versus the if+memmove operation. 

I've also tested this memmove version against my applications that use the frustum tests, as well as small unit test applications that show that the two versions produce the same results. 

Basically the fix compared with the original memmove version is just an additional test to see if there are any bytes to move at all. 

Cheers,
Ola

### CODE REVIEW

**Comment by antarespilot** [comment]
Oops far too many commits in this pull request, was only supposed to be a single one (381eee6)

**Comment by erwincoumans** [comment]
sure, thanks for looking into this!

**Comment by erwincoumans** [comment]
By the way, can you explain how you used the  google benchmark utility exactly?

**Comment by antarespilot** [comment]
I'll write an explanation in the issue #347
